### ISSUE TITLE
Provisioner Chef license accept

### ISSUE THREAD

**Issue by mztriz** [hashicorp/terraform#21350]
I cannot use the latest `Chef Infra Client 15.0.300` with Terraform, it hangs on user acceptance of two product licenses.
```
 provisioner "chef" {
                      ...
                      version = "15.0.300"
                      ...
}


``` 
Acceptance required: 

``` bash
openstack_compute_instance_v2.vm1 (chef): +---------------------------------------------+
openstack_compute_instance_v2.vm1 (chef):             Chef License Acceptance

openstack_compute_instance_v2.vm1 (chef): Before you can continue, 2 product licenses
openstack_compute_instance_v2.vm1 (chef): must be accepted. View the license at
openstack_compute_instance_v2.vm1 (chef): https://www.chef.io/end-user-license-agreement/

openstack_compute_instance_v2.vm1 (chef): Licenses that need accepting:
openstack_compute_instance_v2.vm1 (chef):   * Chef Infra Client
openstack_compute_instance_v2.vm1 (chef):   * Chef InSpec

openstack_compute_instance_v2.vm1 (chef): Do you accept the 2 product licenses (yes/no)?
```

**Comment by apparentlymart** [comment]
Hi @mztriz,

I've not been able to try this myself since I don't a Chef environment ready to test against, but I've head that adding the following configuration will work around this problem:

```
client_options = [ "chef_license 'accept'" ]
```

**Comment by apparentlymart** [comment]
Hi again @mztriz!

I assume your closing this means that the workaround worked! If so, I'm glad. :tada: 

I'm going to reopen this to mark that we ought to do _something_ here: at least, we ought to document that this is how to use Chef after a certain version. We might also consider adding a first-class argument for accepting the license, similar to what's proposed in #21345 for the similar problem in Habitat.

**Comment by pselle** [comment]
Hello! This was released in 0.12.9
https://github.com/hashicorp/terraform/blob/master/CHANGELOG.md#0129-september-17-2019
